src/cascadia/TerminalApp/ActionArgs.h

```diff
@@ -19,15 +19,72 @@
 #include "../../cascadia/inc/cppwinrt_utils.h"
 #include "Utils.h"
 #include "TerminalWarnings.h"
+#include "JsonUtils.h"
 
 // Notes on defining ActionArgs and ActionEventArgs:
 // * All properties specific to an action should be defined as an ActionArgs
 //   class that implements IActionArgs
 // * ActionEventArgs holds a single IActionArgs. For events that don't need
 //   additional args, this can be nullptr.
 
+// Possible Direction values
+// TODO:GH#2550/#3475 - move these to a centralized deserializing place
+JSON_ENUM_MAPPER(::winrt::TerminalApp::Direction)
+{
+    static constexpr std::string_view LeftString{ "left" };
+    static constexpr std::string_view RightString{ "right" };
+    static constexpr std::string_view UpString{ "up" };
+    static constexpr std::string_view DownString{ "down" };
+
+    JSON_MAPPINGS(4) = {
+        pair_type{ LeftString, ::winrt::TerminalApp::Direction::Left },
+        pair_type{ RightString, ::winrt::TerminalApp::Direction::Right },
+        pair_type{ UpString, ::winrt::TerminalApp::Direction::Up },
+        pair_type{ DownString, ::winrt::TerminalApp::Direction::Down },

```

Comment:
is there any point in keeping the strings separate at this point? they could just go right in the `pair_type`s.

Reply:
Yeah. The same can probably be said for ProfileConversionTraits.cpp too